### PR TITLE
Source locator parameter removal when traversing reflection objects

### PR DESCRIPTION
**Pull request by Ocramius**

### CODE REVIEW

File: src/Reflection/ReflectionClass.php [hunk 1]
```diff
@@ -370,22 +380,17 @@ public function getEndLine()
...
         }
 
-        $fqsen = $this->extendsClassType->__toString();
-
-        if (null !== $sourceLocator) {
-            $parent = (new ClassReflector($sourceLocator))->reflect($fqsen);
-        } else {
-            $parent = self::createFromName($fqsen);
-        }
+        // @TODO use actual `ClassReflector` or `FunctionReflector`?
```
Comment (asgrim) [inline src/Reflection/ReflectionClass.php]: is this `@todo` still needed?

File: src/Reflection/ReflectionClass.php [hunk 2]
```diff
@@ -540,18 +544,12 @@ private function getFqsenFromNamedNode(Node\Name $node)
...
+    private function reflectClassForNamedNode(Node\Name $node)
     {
-        $fqsen = $this->getFqsenFromNamedNode($node);
-
-        if (null !== $sourceLocator) {
-            return (new ClassReflector($sourceLocator))->reflect($fqsen);
-        }
-
-        return self::createFromName($fqsen);
+        // @TODO use actual `ClassReflector` or `FunctionReflector`?
```
Comment (asgrim) [inline src/Reflection/ReflectionClass.php]: same

File: src/Reflection/ReflectionClass.php [hunk 3]
```diff
@@ -370,22 +380,17 @@ public function getEndLine()
...
-        $fqsen = $this->extendsClassType->__toString();
-
-        if (null !== $sourceLocator) {
-            $parent = (new ClassReflector($sourceLocator))->reflect($fqsen);
-        } else {
-            $parent = self::createFromName($fqsen);
-        }
+        // @TODO use actual `ClassReflector` or `FunctionReflector`?
+        /* @var $parent self */
+        $parent = $this->reflector->reflect((string) $this->extendsClassType);
```
Comment (asgrim) [inline src/Reflection/ReflectionClass.php]: Can you not just call `$this->reflectClassForNamedNode((string)$this->extendsClassType);`?

File: src/Reflection/ReflectionClass.php (hunk 1 above)
Comment (Ocramius) [inline src/Reflection/ReflectionClass.php]: I am still not sure, that's why it's still there. I have no specific `ClassReflector` hinting, therefore the usage is unsafe in here

File: src/Reflection/ReflectionClass.php (hunk 2 above)
Comment (Ocramius) [inline src/Reflection/ReflectionClass.php]: same as above. `@TODO` stands imo

File: src/Reflection/ReflectionClass.php (hunk 3 above)
Comment (asgrim) [inline src/Reflection/ReflectionClass.php]: Some kind of carnage...

**Pull request merged**